lib/OptimalBranchingCore/src/interfaces.jl

```diff
@@ -16,12 +16,17 @@ Create a branch from the given clause applied to the specified vertices.
 - `vertices`: A vector of vertices to be considered for the branch.
 
 ### Returns
-A tuple of two values:
 - `AbstractProblem`: A new instance of `AbstractProblem` with reduced size.
-- `Number`: The local gain of the branch, which will be added to the global gain.
 """
 function apply_branch end
 
+"""
```

Comment:
How about still use `apply_branch`, but returns an object that contains both the result and the gain.

The interface not computing the gain seems to be not very useful.

Reply:
The function `optimal_branching_rule` use `apply_branch` to see the difference of measure, personally I do not want to send `result_type` into this function.

Reply:
I think the following code is also intuitive.
```julia
measure(apply_branch(problem, candidate.clause, variables).problem, m)
```